jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java

```diff
@@ -0,0 +1,220 @@
+import org.jbehave.core.annotations.BeforeScenario;
+import org.jbehave.core.model.ExamplesTable;
+import org.jbehavesupport.core.TestContext;
+import org.jbehavesupport.core.expression.ExpressionEvaluatingParameter;
+import org.jbehavesupport.core.internal.parameterconverters.ExamplesEvaluationTableConverter;
+import org.jbehavesupport.core.verification.Verifier;
+import org.jbehavesupport.core.verification.VerifierResolver;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.annotation.BeanFactoryAnnotationUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.jbehavesupport.core.internal.ExampleTableConstraints.VERIFIER;
+import static org.jbehavesupport.core.internal.ExamplesTableUtil.convertTable;
+import static org.springframework.util.Assert.isTrue;
+import static org.springframework.util.Assert.notNull;
+
+/**
+ * This class implements steps for testing ssh and provides customization option.
+ * This handler is registered as default. Register our own instance for customization.
+ * <p>
+ * Example:
+ * <pre>
+ * &#064;Configuration
+ * public class MyTestConfiguration {
+ *
+ *     &#064;Bean
+ *     public CustomSshHandler customSshHandler() {
+ *         return new CustomSshHandler();
+ *     }
+ *
+ * }
+ *
+ * public class CustomSshHandler extends SshHandler {
+ *
+ *     //project specific customization
+ *
+ * }
+ * </pre>
+ */
+
+@Slf4j
+@RequiredArgsConstructor
+public class SshHandler {
+    private final TestContext testContext;
+    private final  ConfigurableListableBeanFactory beanFactory;
+    private final  ExamplesEvaluationTableConverter tableConverter;
+    private final  VerifierResolver verifierResolver;
+
+
+    @Value("${ssh.max.assert.count:10}")
+    private int maxSoftAssertCount;
+
+    private ZonedDateTime scenarioStart;
+    private ZonedDateTime logStartTime;
+    private ZonedDateTime logEndTime;
+
+    private MultiKeyMap<MultiKey, String> logCache = MultiKeyMap.multiKeyMap(new LRUMap());
+
+    @BeforeScenario
+    public void init() {
+        scenarioStart = ZonedDateTime.now();
+        logStartTime = null;
+        logEndTime = null;
+        logCache.clear();
+    }
+
+    public void markLogTime(String logTimeAlias) {
+        testContext.put(logTimeAlias, ZonedDateTime.now());
+    }
+
+    public void saveLogStartTime(){
+        logStartTime = ZonedDateTime.now();
+    }
+
+    public void saveLogEndTime(){
+        logEndTime = ZonedDateTime.now().plusSeconds(1);
+    }
+
+    public void saveLogStartTimeOnSaved(ExpressionEvaluatingParameter<String> contextAlias) {
```

Comment:
Minor but should probably be setLogStartTimeOnSaved?

```suggestion
    public void setLogStartTimeOnSaved(ExpressionEvaluatingParameter<String> contextAlias) {
```